meson.build

```diff
@@ -0,0 +1,239 @@
+config_h.set_quoted('GETTEXT_PACKAGE', proj_name)
+config_h.set_quoted('PACKAGE', proj_name)
+config_h.set_quoted('PACKAGE_BUGREPORT', proj_bugreport)
+config_h.set_quoted('PACKAGE_NAME', proj_name)
+config_h.set_quoted('PACKAGE_STRING', '@0@ @1@'.format(proj_name, proj_version))
+config_h.set_quoted('PACKAGE_TARNAME', proj_name)
+config_h.set_quoted('PACKAGE_VERSION', proj_version)
+config_h.set_quoted('LOCALEDIR', full_localedir)
+
+# Check for library functions
+intl_dep = cc.find_library('intl', required : false)
+m_dep = cc.find_library('m', required : false)
+if cc.sizeof('uint16_t', prefix : '#include <stdint.h>') == -1
+  error('uint16_t is not defined.')
+endif
+foreach f : ['setlocale', 'strchr', 'floor', 'pow']
+  if not cc.has_function(f, dependencies : m_dep)
+    error('Function "@0@" is not available.'.format(f))
+  endif
+endforeach
+
+# NLS
+opt_nls = get_option('nls')
+if not opt_nls.disabled()
+  have_gettext_func = cc.has_function('gettext', dependencies : intl_dep)
+  if not have_gettext_func and opt_nls.enabled()
+    error('Function gettext() is not available.')
+  endif
+  have_libintl_h = cc.has_header('libintl.h')
+  if not have_libintl_h and opt_nls.enabled()
+    error('libintl.h is not installed.')
+  endif
+  if have_gettext_func and have_libintl_h
+    config_h.set('ENABLE_NLS', 1)
+  endif
+endif
+
+# Check for header files
+system_headers = [
+  'locale.h',
+  'signal.h',
+  'stdint.h',
+  'stdlib.h',
+  'string.h',
+  'unistd.h',
+]
+foreach h : system_headers
+  if cc.has_header(h)
+    config_h.set('HAVE_@0@'.format(h.to_upper().underscorify()), 1)
+  endif
+endforeach
+
+# Build configuration
+config_build = configuration_data()
+
+null_dep = disabler()
+
+# Check DRM method
+drm_dep = dependency('libdrm', required : get_option('drm'))
+if drm_dep.found()
+  config_h.set('ENABLE_DRM', 1)
+endif
+
+# Check RANDR method
+xcb_dep = null_dep
+xcb_randr_dep = dependency('xcb-randr', required : get_option('randr'))
+if xcb_randr_dep.found()
+  xcb_dep = dependency('xcb')
+endif
+if xcb_dep.found()
+  config_h.set('ENABLE_RANDR', 1)
+endif
+
+# Check VidMode method
+x11_dep = null_dep
+xxf86vm_dep = dependency('xxf86vm', required : get_option('vidmode'))
+if xxf86vm_dep.found()
+  x11_dep = dependency('x11')
+endif
+if x11_dep.found()
+  config_h.set('ENABLE_VIDMODE', 1)
+endif
+
+# Check Quartz (macOS) method
+appservices_dep = dependency('appleframeworks',
+  modules : 'applicationservices', required : get_option('quartz'))
+if appservices_dep.found() and cc.has_header(
+    'ApplicationServices/ApplicationServices.h')
+  config_h.set('ENABLE_QUARTZ', 1)
+endif
+
+# Check Windows GDI method
+gdi32_dep = cc.find_library('gdi32', required : get_option('wingdi'))
+if gdi32_dep.found() and cc.has_header('windows.h')
+  config_h.set('ENABLE_WINGDI', 1)
+endif
+
+# Check Geoclue2 location provider
+gio_dep = dependency(
+  'gio-2.0', version : '>= 2.26', required : get_option('geoclue2'))
+if gio_dep.found()
```

Comment:
if the geoclue2 option is disabled, but the gio dependency is found, what happens? same for similar checks in this file.

Reply:
When `-Dgeoclue2=disabled` is specified, *Geoclue2 support will be disabled and GIO will not be used* (GIO is used only in Geoclue2 location provider).

* The value of the `geoclue2` option (`type : feature`) is `auto` in `meson_options.txt`: Geoclue2 support will be enabled only when GIO is detected
* If `-Dgeoclue2=enabled` is specified and GIO is not found, an error will occur (build failure)